src-tauri/src/swarm.rs

```diff
@@ -74,59 +74,88 @@ pub fn is_swarm_mode_running() -> bool {
     return false;
 }
 
+#[tauri::command(async)]
 pub fn create_environment(handle: tauri::AppHandle) -> String {
-    // Get the application data directory
-    let app_data_dir = tauri::api::path::home_dir().expect("🙈 Failed to get app data directory");
-    let app_data_dir = app_data_dir
-        .join(".config/prem");
-    let app_data_dir = app_data_dir
-        .to_str()
-        .expect("🙈 Failed to convert app data dir path to str");
-
-    // Get create env path
-    let binding = handle.path_resolver()
-        .resolve_resource("petals")
-        .expect("🙈 Failed to find `create_env.sh`");
-    let petals_path = binding
-        .to_str()
-        .expect("🙈 Failed to convert petals path to str");
-
-    let python = format!("{app_data_dir}/miniconda/envs/prem_app/bin/python");
-
-    // Set env variables
-    let mut env = HashMap::new();
-    env.insert("PREM_APPDIR".to_string(), app_data_dir.to_string());
-    env.insert("PREM_PYTHON".to_string(), python.clone());
-
-    // Run the bash script
+    println!("🐍 Creating the environment...");
+    let petals_path = get_petals_path(handle);
+    let env = get_environment();
     let _ = Command::new("sh")
         .args([format!("{petals_path}/create_env.sh")])
-        .envs(env)
+        .envs(env.clone())
         .output()
         .expect("🙈 Failed to create env");
-    python
+    env.get(&String::from("PREM_PYTHON")).unwrap().to_string()
 }
 
 
 #[tauri::command(async)]
-pub fn run_swarm_mode(handle: tauri::AppHandle, num_blocks: i32, model: String, public_name: String){
-    let python: String = create_environment(handle);
-    println!("🚀 Starting the Swarm...");
+pub fn install_swarm_requirements(handle: tauri::AppHandle) {
+    let petals_path = get_petals_path(handle);
+    let env = get_environment();
+    let python = env.get(&String::from("PREM_PYTHON")).unwrap().to_string();
+    let pip = python.replace("python", "pip");
+    let requirements = format!("{petals_path}/requirements.txt");
+    println!("🐍 Installing the requirements with {} install -r {}...", pip, requirements);
+    let _ = Command::new(&pip)
+        .args([
+            "install",
+            "-r",
+            &requirements
+        ])
+        .output()
+        .expect("🙈 Failed to install requirements");
```

Comment:
```suggestion
    let exit_status = Command::new(&pip)
        .args([
            "install",
            "-r",
            &requirements
        ])
        .spawn()
        .expect("🙈 Couldn't start `pip install -r $PETALS_PATH/requirements.txt`")
        .wait()
        .expect("🙈 Didn't run `pip install -r $PETALS_PATH/requirements.txt");
    if !exit_status.success() {
        eprintln!("🙈 Failed to `pip install -r $PETALS_PATH/requirements.txt");
    }
```

Output returns an error only if the command failed to execute, the command itself could still fail and you won't report or break on it.


Also I recommend using Result over directly using `expect` it's essentially the same as unwrap so it will crash the app if the command fails.

Reply:
@swarnimarun Command is a tauri [Command](https://docs.rs/tauri/latest/tauri/api/process/struct.Command.html) and does not have `wait()`

Reply:
Removed install requirements